gnomad/utils/annotations.py

```diff
@@ -2312,21 +2325,55 @@ def _create_group_dicts(
         "cohort": {"id": "ALL"},
     }
 
-    # Create ancillaryResults for additional frequency and popMaxFAF95 information
+    # Check allele balance for heterozygotes values.
+    # Flagged allele balance values are those in bins > 0.90.
+    # Each bin is 0.05, so flagged values are in the last 2 bins.
+    if len(input_struct.ab_hist_alt.bin_freq) != 20:
+        raise ValueError(
+            f"{gnomad_id} ab_hist_alt.bin_freq had "
+            f"{len(input_struct.ab_hist_alt.bin_freq)} items, expected 20"
+        )
+    # The bin_freq should be in order but we can verify the order from bin_edges.
+    ab_bin_freq = list(
+        map(
+            lambda x: x[1],
+            sorted(
+                zip(
+                    input_struct.ab_hist_alt.bin_edges,
+                    input_struct.ab_hist_alt.bin_freq,
+                ),
+                key=lambda x: x[0],
+            ),
+        )
+    )
+
+    # Create ancillaryResults for additional frequency and popMaxFAF95 information.
     ancillaryResults = {
         "homozygotes": overall_freq["homozygote_count"],
-        "popMaxFAF95": {
-            "frequency": input_struct.faf95.popmax,
-            "confidenceInterval": 0.95,
-        },
+        "qcFilters": list(input_struct.filters),
+        "lowComplexityRegion": input_struct.region_flag.lcr,
+        "heterozygousSkewedAlleleCount": sum(ab_bin_freq[-2:]),
     }
 
+    # Add hemizygote count if not autosomal or PAR.
+    if not input_struct.in_autosome_or_par:
+        hemizygote_count = input_struct.freq[freq_index_dict["XY-adj"]].AC
+        ancillaryResults["hemizygotes"] = hemizygote_count
+
     if input_struct.faf95.popmax_population is not None:
-        ancillaryResults["popMaxFAF95"][
-            "popFreqID"
-        ] = f"{gnomad_id}.{input_struct.faf95.popmax_population.upper()}"
+        grpMaxFAF95 = (
+            {
+                "frequency": input_struct.faf95.popmax,
+                "confidenceInterval": 0.95,
+                "grpFreqId": (
+                    f"{gnomad_id}.{input_struct.faf95.popmax_population.upper()}"
+                ),
+            },
+        )
```

Comment:
@klaricch this is the added code (in this region) to fix their new schema for accepting NONE types - lmk if this looks right